### PR TITLE
#173977481 Feature enable delete lost or found items

### PR DESCRIPTION
**Pull request by SimonImanigirimpuhwe**
### What does this PR do?

- Allow users to delete found or lost items from database

### Description of Task to be completed?

- Setting a route to handle delete items request.

- Send a `Delete` request to remove document from database.

### How should this be manually tested?

- Clone a repo `git clone`

- Then switch a branch by ` git checkout ft-delete-items-#173977481` .

- Run `npm install` to install all dependancies. And then run `npm run dev` to start the server.

- In postman send a delete request on `http://localhost:3000/api/v1/items/delete/_id` with a valid `_id` you should see a deleted document with message including the document number. Otherwise you should see an error indicating the nonexistence of document. 

### Any background context you want to provide?

- To be able to send a delete request you must login and provide token in header key as `auth-token` and `token` as value.

### What are the relevant pivotal tracker stories?
#173977481

### CODE REVIEW

File: src/controllers/lostDoc/deleteItem.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import DeleteItem from '../../models/lost';
+
+export default async (req, res) =>{
+    try{
+        const deleteDoc = await DeleteItem.findByIdAndDelete(req.params._id)
+
+        return res.status(200).send({msg:`Document with number ${deleteDoc.documentNumber} deleted successfully`, doc});
+    }catch(error){
+        return res.status(400).send({error: 'No document with such Id found'});
```
Comment (Habinezajanvier) [inline src/controllers/lostDoc/deleteItem.js]: You have put every error in catch, If possible extract not found error and fix it in try not in catch

File: src/app.js [hunk 2]
```diff
@@ -19,7 +18,6 @@ app.use('/api/v1/items', routes);
 app.use('/api/v1/users', emailAuthRoutes);
 app.use('/api/v1/users/auth', socialAuthRoutes);
 app.use('/api/v1/users/profile', profileRouter);
-app.use('/api/v1/items/search', searchRouter);
```
Comment (Habinezajanvier) [inline src/app.js]: Hey @SimonImanigirimpuhwe , You have removed the line that enable search items,

File: src/controllers/lostDoc/deleteItem.js [hunk 3]
```diff
@@ -0,0 +1,11 @@
+import DeleteItem from '../../models/lost';
+
+export default async (req, res) =>{
+    try{
+        const deleteDoc = await DeleteItem.findByIdAndDelete(req.params._id)
+
+        return res.status(200).send({msg:`Document with number ${deleteDoc.documentNumber} deleted successfully`, doc});
```
Comment (Habinezajanvier) [inline src/controllers/lostDoc/deleteItem.js]: Remove this undefined variable on this line

File: src/app.js (hunk 2 above)
Comment (SimonImanigirimpuhwe) [reply src/app.js]: It was removed unintentionally, resolved issue to be committed asap

File: src/controllers/lostDoc/deleteItem.js (hunk 1 above)
Comment (SimonImanigirimpuhwe) [reply src/controllers/lostDoc/deleteItem.js]: lemme take care of it

**Review by Habinezajanvier — APPROVED** [review]
Good

**Pull request merged**